backend/apps/progress/views.py

```
@@ -1,1515 +1,1609 @@
+
+        if start_param:
+            try:
+                start_date = datetime.datetime.strptime(start_param, "%Y-%m-%d").date()
+            except ValueError:
+                start_date = timezone.localdate() - datetime.timedelta(days=365)
+        else:
+            start_date = timezone.localdate() - datetime.timedelta(days=365)
+
+        if end_param:
+            try:
+                end_date = datetime.datetime.strptime(end_param, "%Y-%m-%d").date()
```

Ruff (0.15.21)

[warning] Naive datetime constructed using `datetime.datetime.strptime()` without %z

(DTZ007)